unpack-trees.c

```diff
@@ -1581,6 +1581,9 @@ int unpack_trees(unsigned len, struct tree_desc *t, struct unpack_trees_options
 	o->merge_size = len;
 	mark_all_ce_unused(o->src_index);
 
+	if (o->src_index->fsmonitor_last_update)
+		o->result.fsmonitor_last_update = o->src_index->fsmonitor_last_update;
+
```

Comment:
This probably makes sense on its own, in a separate patch (which I would move to the front of the PR, if I were you). It should not break any tests, or does it?